### PR TITLE
[FLINK-7205] [table] Add UUID supported in SQL and TableApi

### PR DESCRIPTION
**Pull request by buptljy**
## What is the purpose of the change
* Add UUID supported in SQL and TableApi.
## Brief change log
* Add UUID function.
## Verifying this change
* Unit tests.

## Documentation
* add in table.md and sql.md

### CODE REVIEW

**Comment by yanghua** [comment]
+1

**Comment by yanghua** [comment]
@buptljy conflicts, please update and rebase this PR~

**Comment by buptljy** [comment]
@yanghua fix.

File: docs/dev/table/tableApi.md [hunk 1]
```diff
@@ -2311,6 +2311,17 @@ randInteger(seed integer, bound integer)
     </td>
    </tr>
 
+    <tr>
+     <td>
+       {% highlight java %}
```
Comment (yanghua) [inline docs/dev/table/tableApi.md]: scala doc also needed~

Comment (buptljy) [reply docs/dev/table/tableApi.md]: Sorry I forgot. It's already added now.

**Comment by buptljy** [comment]
@hequn8128 Thank you for your suggestions. I've fixed most of the problems. But for the nonDeterministic cases, I don't think we're able to test them with what we have now because we need two rows in test data to verify their distinctions.

**Comment by hequn8128** [comment]
@buptljy  I don't agree. Without a test, we don't know whether the bug has been fixed.
As a choice you can use `NULLIF(uuid(), uuid()) IS NOT NULL` to test, i.e., the result should return true.

**Comment by buptljy** [comment]
@hequn8128 
First, NULLIF(uuid(), uuid()) won't succeed to return null because of the "ReduceExpressionsRule", which will regard uuid() and uuid() as the same based on the "uuid()" string, it's the same for othere functions like current_timestamp.
Second, putting two uuid() into the same row may not test "isDeterministic" function. for example, we assume that the "isDeterministic" returns true: the outputs are different when the input is  "select uuid(),uuid() from testdata(1 row)", while the outputs are the same when the input is "select uuid() from testdata(2 rows)".

**Comment by hequn8128** [comment]
@buptljy Ok, you are right.
How about add a test in `CalcITCase` in package of `org.apache.flink.table.runtime.stream.table`. 

```
  @Test
  def testUDFNotDeterministic(): Unit = {
    val env = StreamExecutionEnvironment.getExecutionEnvironment
    val tEnv = TableEnvironment.getTableEnvironment(env)
    StreamITCase.testResults = mutable.MutableList()
    val ds = StreamTestData.getSmall3TupleDataStream(env).toTable(tEnv).select(uuid())

    val results = ds.toAppendStream[Row]
    results.print()
    results.addSink(new StreamITCase.StringSink[Row])
    env.execute()

    assert(StreamITCase.testResults.distinct.size > 1)
  }
```

**Comment by buptljy** [comment]
@hequn8128 Good idea! it's added now.

File: flink-libraries/flink-table/src/test/scala/org/apache/flink/table/runtime/stream/table/CalcITCase.scala [hunk 2]
```diff
@@ -379,4 +379,19 @@ class CalcITCase extends AbstractTestBase {
...
+
+  @Test
+  def testUDFNotDeterministic(): Unit = {
+    val env = StreamExecutionEnvironment.getExecutionEnvironment
+    val tEnv = TableEnvironment.getTableEnvironment(env)
+    StreamITCase.testResults = mutable.MutableList()
+    val ds = StreamTestData.getSmall3TupleDataStream(env).toTable(tEnv).select(uuid())
+
+    val results = ds.toAppendStream[Row]
+    results.print()
```
Comment (hequn8128) [inline flink-libraries/flink-table/src/test/scala/org/apache/flink/table/runtime/stream/table/CalcITCase.scala]: remove this line. = =!

**Comment by buptljy** [comment]
The error appears in travis-ci is not caused by this PR.

**Comment by buptljy** [comment]
@twalthr Thanks! I've already fixed them.

**Comment by twalthr** [comment]
Thank you @buptljy. Merging...

**Pull request closed without merging**